sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -316,10 +334,98 @@ private Message createManagementMessage(String operation, String linkName) {
+     * @throws AmqpException When payload exceeded maximum message allowed size.
+     */
+    private Map<String, Object> createScheduleMessgeAmqpValue(ServiceBusMessage messageToSchedule, int maxMessageSize) {
+
+        Message message = messageSerializer.serialize(messageToSchedule);
+
+        // The maxsize allowed logic is from ReactorSender, this logic should be kept in sync.
+        final int payloadSize = messageSerializer.getSize(message);
+        final int allocationSize =
+            Math.min(payloadSize + MAX_MESSAGING_AMQP_HEADER_SIZE_BYTES, maxMessageSize);
+        final byte[] bytes = new byte[allocationSize];
+
+        int encodedSize;
+        try {
+            encodedSize = message.encode(bytes, 0, allocationSize);
+        } catch (BufferOverflowException exception) {
+            final String errorMessage =
+                String.format(Locale.US,
+                    "Error sending. Size of the payload exceeded maximum message size: %s kb",
+                    maxMessageSize / 1024);
+            throw logger.logExceptionAsWarning(new AmqpException(false,
+                AmqpErrorCondition.LINK_PAYLOAD_SIZE_EXCEEDED, errorMessage, exception, getErrorContext()));
+        }
+        HashMap<String, Object> messageEntry = new HashMap<>();
+        messageEntry.put(MESSAGE, new Binary(bytes, 0, encodedSize));
+        messageEntry.put(MESSAGE_ID, message.getMessageId());
+
+        Collection<HashMap<String, Object>> messageList = new LinkedList<>();
+        messageList.add(messageEntry);
+
+        Map<String, Object> requestBodyMap = new HashMap<>();
+        requestBodyMap.put(MESSAGES, messageList);
+        return requestBodyMap;
+    }
+
     private AmqpErrorContext getErrorContext() {
         return new SessionErrorContext(fullyQualifiedNamespace, entityPath);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Void> cancelScheduledMessage(long sequenceNumber) {
+        return isAuthorized(CANCEL_SCHEDULED_MESSAGE_OPERATION).thenMany(createRequestResponse.flatMap(channel -> {
+
+            Message requestMessage = createManagementMessage(CANCEL_SCHEDULED_MESSAGE_OPERATION,
+                channel.getReceiveLinkName());
+
+            requestMessage.setBody(new AmqpValue(Collections.singletonMap(SEQUENCE_NUMBERS,
+                new Long[]{sequenceNumber})));
+            return channel.sendWithAck(requestMessage);
+        }).map(responseMessage -> {
+            int statusCode = RequestResponseUtils.getResponseStatusCode(responseMessage);
+
+            if (statusCode == AmqpResponseCode.OK.getValue()) {
+                return Mono.empty();
+            }
+            return Mono.error(new AmqpException(false, "Could not cancel scheduled message with sequence number "
+                + sequenceNumber, getErrorContext()));
+        })).then();
+    }
+
+    private Flux<Long> scheduleMessage(ServiceBusMessage messageToSchedule, Instant scheduledEnqueueTime,
```

Comment:
Don't think this needs to be its own function. It's only used once. Same with your helper method to create a message. Functions are pieces of code that can be reused.